### PR TITLE
Documentation only, nep

### PR DESCRIPTION
**Pull request by juancarlospaco**
- Documentation only, tiny diff.
- Obvious suggestions, like do not use `float` for money, etc.

### CODE REVIEW

File: doc/nep1.rst [hunk 1]
```diff
@@ -247,6 +247,10 @@ Coding Conventions
   variables remain immutable, and gives those who read the code a better idea
   of the code's purpose.
 
+- Do not use `float` for money.
+
+- Do not use `Natural` or `Positive` as return type, use `int` or `uint` as return type instead.
```
Comment (timotheecour) [inline doc/nep1.rst:252]: araq gave the rationale for that somewhere, please find it and add it so that this rule doesn't appear arbitrary

File: doc/nep1.rst [hunk 2]
```diff
@@ -247,6 +247,10 @@ Coding Conventions
   variables remain immutable, and gives those who read the code a better idea
   of the code's purpose.
 
+- Do not use `float` for money.
```
Comment (timotheecour) [inline doc/nep1.rst:250]: add rationale even if obvious for most; but also, I'm not sure if this should be mentioned at all, seems out of scope

File: doc/nep1.rst [hunk 3]
```diff
@@ -247,6 +247,10 @@ Coding Conventions
...
+
```
Comment (timotheecour) [inline doc/nep1.rst:251]: these belong somewhere else, not in nep1.rst; maybe https://nim-lang.github.io/Nim/contributing.html#best-practices

Comment (juancarlospaco) [reply doc/nep1.rst:251]: ok
